feat: add support for enroot as container runtime

ffais opened the pull request:
# Summary

Add support for enroot as container runtime.

---

<!-- Add, if any, the related issue here, e.g. #6 -->

**Related issue :** #76

dciangot commented on docker/Dockerfile-debug:
```diff
@@ -0,0 +1,70 @@
+FROM ubuntu:22.04
```
If you have some notes on how to use this for developing the plugin it would be very nice to add it in a .md file

dciangot commented on pkg/slurm/Create.go:
```diff
@@ -166,32 +141,21 @@ func (h *SidecarHandler) SubmitHandler(w http.ResponseWriter, r *http.Request) {
+		commstr1 := prepareRuntimeCommand(h.Config, container, metadata)
+		log.G(h.Ctx).Debug("-- Appending all commands together...")
+		runtime_command := append(commstr1, envs...)
+		switch h.Config.ContainerRuntime {
```
It's ok defaulting to singularity, I wonder if maybe we can rise an fatal error, earlier in the code, if the runtime name is not singularity neither enroot. 

It can happen that one user blindly put in the config a runtime that is not in that list.

dciangot reviewed (commented):
thanks @ffais . It looks very promising, and definetely in line with what we discussed last time. 

I added a few comments that should be not to heavy to implement! If you have any question, please ask!

ffais commented:
Hi @dciangot, I'm finally back to work on this PR. I rebased the PR with main adding your latest changes, added the improvements you suggestet and a few small fixes. There is still some room for improvements in volume management and support for health checks on enroot-side. Let me know what you think.

dciangot commented:
hi @ffais thanks for the awesome work! I'm reviewing the PR, but I think that overall looks like ready to be evaluated for the merge. If there is any enhancement that you plan for the enroot part I'd go with separate PR, so not to delay this one. Sounds good for you?

dciangot reviewed (approved):
LGTM. Please open issues for whatever enroot volume management improvement.

dciangot commented:
also! @ffais please open an issue to create test cases for enroot in interlink main repo! Not a requirement for this to be merged, but useful to keep enroot working later on

dciangot commented:
@ffais as soon as you confirm the related issues are created I'll go ahead with the PR inclusion!
